### PR TITLE
New Resource: PublicDnsNamespace

### PR DESCRIPTION
**Pull request by atsushi-ishibashi**

### CODE REVIEW

**Comment by atsushi-ishibashi** [comment]
```
TF_ACC=1 go test ./aws -v -run=TestAccAwsServiceDiscoveryPublicDnsNamespace_basic -timeout 120m
=== RUN   TestAccAwsServiceDiscoveryPublicDnsNamespace_basic
--- PASS: TestAccAwsServiceDiscoveryPublicDnsNamespace_basic (112.03s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	112.077s
```

**Comment by atsushi-ishibashi** [comment]
Ready to review👍

File: aws/resource_aws_service_discovery_public_dns_namespace_test.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+func TestAccAwsServiceDiscoveryPublicDnsNamespace_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsServiceDiscoveryPublicDnsNamespaceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccServiceDiscoveryPublicDnsNamespaceConfig(acctest.RandString(5)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsServiceDiscoveryPublicDnsNamespaceExists("aws_service_discovery_public_dns_namespace.test"),
```
Comment (radeksimko) [inline aws/resource_aws_service_discovery_public_dns_namespace_test.go]: Can you also add 2 more checks here for the computed fields, please? i.e.

```go
resource.TestCheckResourceAttrSet("aws_service_discovery_public_dns_namespace.test", "arn"),
resource.TestCheckResourceAttrSet("aws_service_discovery_public_dns_namespace.test", "hosted_zone"),
```

File: aws/resource_aws_service_discovery_public_dns_namespace.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+	input := &servicediscovery.CreatePublicDnsNamespaceInput{
+		Name: aws.String(d.Get("name").(string)),
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		input.Description = aws.String(v.(string))
+	}
+
+	resp, err := conn.CreatePublicDnsNamespace(input)
```
Comment (radeksimko) [inline aws/resource_aws_service_discovery_public_dns_namespace.go]: Shouldn't we also use `CreatorRequestId` to make it safe to create multiple resources in parallel?

We usually use randomly generated string, like here:
https://github.com/terraform-providers/terraform-provider-aws/blob/0c0887cff37a40c1d35ee2bbe1788831d1f871e2/aws/resource_aws_mq_broker.go#L184

File: aws/resource_aws_service_discovery_public_dns_namespace.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+		return err
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{servicediscovery.OperationStatusSubmitted, servicediscovery.OperationStatusPending},
+		Target:     []string{servicediscovery.OperationStatusSuccess},
+		Refresh:    servicediscoveryOperationRefreshStatusFunc(conn, *resp.OperationId),
+		Timeout:    5 * time.Minute,
+		Delay:      10 * time.Second,
+		MinTimeout: 3 * time.Second,
```
Comment (radeksimko) [inline aws/resource_aws_service_discovery_public_dns_namespace.go]: Is there any particular reason for delaying the refresh here? i.e. do the two fields `Delay` & `MinTimeout` have any reason? Based on my brief testing the deletion can finish in 2 seconds, so we're potentially letting  the user wait for 8 more seconds unnecessarily here.

File: aws/resource_aws_service_discovery_public_dns_namespace.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
```
Comment (radeksimko) [inline aws/resource_aws_service_discovery_public_dns_namespace.go]: Is there any particular reason for delaying the refresh here? i.e. do the two fields `Delay` & `MinTimeout` have any reason? Based on my brief testing the creation can finish in 2 seconds, so we're potentially letting  the user wait for 8 more seconds unnecessarily here.

File: aws/resource_aws_service_discovery_public_dns_namespace.go [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	d.Set("description", resp.Namespace.Description)
+	d.Set("arn", resp.Namespace.Arn)
+	if resp.Namespace.Properties != nil {
+		d.Set("hosted_zone", resp.Namespace.Properties.DnsProperties.HostedZoneId)
```
Comment (radeksimko) [inline aws/resource_aws_service_discovery_public_dns_namespace.go]: I wonder if Amazon has plans for expanding `Namespace.Properties` or `Namespace.Properties.DnsProperties`.

It's no big deal though as we can always deprecate this 1st-level field in favour of its nested version if it becomes necessary.

**Comment by atsushi-ishibashi** [comment]
@radeksimko Ok👍
```
TF_ACC=1 go test ./aws -v -run=TestAccAwsServiceDiscoveryPublicDnsNamespace_basic -timeout 120m
=== RUN   TestAccAwsServiceDiscoveryPublicDnsNamespace_basic
--- PASS: TestAccAwsServiceDiscoveryPublicDnsNamespace_basic (65.66s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	65.742s
```

**Pull request merged**